src/keysignature.ts

```diff
@@ -32,27 +33,37 @@ export class KeySignature extends StaveModifier {
 
     this.setKeySig(keySpec, cancelKeySpec, alterKeySpec);
     this.setPosition(StaveModifierPosition.BEGIN);
-    this.glyphs = [];
+    this.children = [];
     this.paddingForced = false;
   }
 
   // Add an accidental glyph to the `KeySignature` instance which represents
   // the provided `acc`. If `nextAcc` is also provided, the appropriate
   // spacing will be included in the glyph's position
-  convertToGlyph(acc: { type: string; line: number }, nextAcc: { type: string; line: number }, stave: Stave): void {
+  protected convertToGlyph(
+    acc: { type: string; line: number },
+    nextAcc: { type: string; line: number },
+    stave: Stave
+  ): void {
     const code = Tables.accidentalCodes(acc.type);
     const glyph = new Element(Category.KeySignature);
     glyph.setText(code);
 
     // Determine spacing between current accidental and the next accidental
-    const extraWidth = 1;
+    let extraWidth = 1;
     // Place the glyph on the stave
     glyph.setYShift(stave.getYForLine(acc.line));
-    if (this.glyphs.length > 0) {
-      const prevGlyph = this.glyphs[this.glyphs.length - 1];
+    if (this.children.length > 0) {
+      const prevGlyph = this.children[this.children.length - 1];
+      if (
+        (prevGlyph.getText() === Glyphs.accidentalNatural || glyph.getText() === Glyphs.accidentalNatural) &&
+        Math.abs(glyph.getYShift() - prevGlyph.getYShift()) < 10
+      ) {
+        extraWidth = 2;
+      }
```

Comment:
The conditional is a bit difficult to read.

Can we extract them out into temporarly local variables?

Something like:
```
const prevGlyphIsNatural = prevGlyph.getText() === Glyphs.accidentalNatural;
const currGlyphIsNatural = glyph.getText() === Glyphs.accidentalNatural;
const glyphsAreTooClose = Math.abs(glyph.getYShift() - prevGlyph.getYShift()) < 10;
```

I don't know what are the best variable names here. You can choose.

Reply:
I have defined to local functions: `isNatural` and `yShiftDiff`